src/app/account.service.ts

```diff
@@ -74,6 +83,47 @@ export class AccountService {
     }
   }
 
+  getDerivedUser(publicKey: string, blockHeight: number): DerivedUserInfo{
+    const privateUser = this.getPrivateUsers()[publicKey];
+
+
+    this.entropyService.setNewTemporaryEntropy();
+    const derivedMnemonic = this.entropyService.temporaryEntropy?.mnemonic;
+    const derivedKeychain = this.cryptoService.mnemonicToKeychain(derivedMnemonic);
+    const derivedSeedHex = this.cryptoService.keychainToSeedHex(derivedKeychain);
+    const derivedPrivateKey = this.cryptoService.seedHexToPrivateKey(derivedSeedHex);
+    const derivedPublicKey = this.cryptoService.privateKeyToDeSoPublicKey(derivedPrivateKey, privateUser.network);
+
+    // By default we authorize this derived key for 10,000 blocks.
+    const expirationBlock = blockHeight + 10000;
+
+    const expirationBlockBuffer = uint64ToBufBigEndian(expirationBlock);
+    const derivedPublicKeyBuffer = derivedPrivateKey.getPublic().encode('array', true);
+    const accessHash = sha256.x2([...derivedPublicKeyBuffer, ...expirationBlockBuffer]);
+    const accessSignature = this.signingService.signBurn(privateUser.seedHex, [accessHash])[0];
+
+    // We compute an owner JWT with a month-long expiration. This is needed for some backend endpoints.
+    const keyEncoder = new KeyEncoder('secp256k1');
+    const encodedPrivateKey = keyEncoder.encodePrivate(privateUser.seedHex, 'raw', 'pem');
+    const jwt = jsonwebtoken.sign({ }, encodedPrivateKey, { algorithm: 'ES256', expiresIn: '30 days' });
+
+    // We compute a derived key JWT with a month-long expiration. This is needed for shared secret endpoint.
+    const encodedDerivedPrivateKey = keyEncoder.encodePrivate(derivedSeedHex, 'raw', 'pem');
+    const derivedJwt = jsonwebtoken.sign({ }, encodedDerivedPrivateKey, { algorithm: 'ES256', expiresIn: '30 days' });
+
+    return {
+      btcDepositAddress: privateUser.btcDepositAddress,
```

Comment:
Won't they need a new, derived `btcDepositAddress` and `ethDepositAddress` so they can actually sign these transactions with their derived key?

Reply:
We need to consider how we treat btc & eth deposit addresses for derived keys in core/backend.